### PR TITLE
Initialize Event Recorder for executer

### PR DESCRIPTION
**Pull request by razo7**
- Initialization was missing, and [`e.recorder`](https://github.com/medik8s/fence-agents-remediation/blob/main/pkg/cli/cliexecuter.go#L187) was nil.
- Fix unit-tests

Fix [ECOPROJECT-1816](https://issues.redhat.com//browse/ECOPROJECT-1816)

### CODE REVIEW

**Comment by mshitrit** [comment]
I think a UT capturing an event would be nice

File: controllers/fenceagentsremediation_controller_test.go [hunk 1]
```diff
@@ -51,6 +52,10 @@ const (
 	timeoutPreRemediation  = "1s" // this timeout is used for the other steps that occur before remediation is completed
 	timeoutPostRemediation = "2s" // this timeout is used for the other steps that occur after remediation is completed
 	pollInterval           = "200ms"
+
+	// eventSteps
+	eventExixst    = "Verifying that event %s was created from"
+	eventNotExixst = "Verifying that event %s was not created from"
```
Comment (mshitrit) [inline controllers/fenceagentsremediation_controller_test.go:58]: Typo: Exixst

File: controllers/controllers_suite_test.go [hunk 2]
```diff
@@ -48,12 +48,13 @@ import (
...
-	ctx          context.Context
-	cancel       context.CancelFunc
-	fakeRecorder *record.FakeRecorder
+	k8sClient             client.Client
+	k8sManager            manager.Manager
+	testEnv               *envtest.Environment
+	ctx                   context.Context
+	cancel                context.CancelFunc
+	fakeReconcileRecorder *record.FakeRecorder
+	fakeExecRecorder      *record.FakeRecorder
```
Comment (mshitrit) [inline controllers/controllers_suite_test.go:57]: I think it's fine to use the same recorder in the test code, it'll simplify the code significantly and I don't think there is a major trade-off as far as test quality.

Comment (razo7) [reply controllers/controllers_suite_test.go:57]: Indeed. I was not sure what is better

**Pull request merged**